artifact/metadata.go

```diff
@@ -74,7 +74,7 @@ func (i *Info) Write(p []byte) (n int, err error) {
 // UpdateType provides information about the type of update.
 // At the moment the only built-in type is "rootfs-image".
 type UpdateType struct {
-	Type string `json:"type"`
+	Type *string `json:"type"`
```

Comment:
The changes to the artifact format need to be documented in `Documentation/artifact-format-v3.md`. Document that `type` can be `null`, and that this has the special properties:

* `data/xxxx.tar.gz` must be missing or empty.
* No `meta-data`.
* No augmented artifacts.